### DISCUSSION TITLE
Access to requested resource is denied

### DISCUSSION THREAD

**Discussion by user6116** [abuzuhri/Amazon-SP-API-CSharp#81] [General]
Hi there.  First I wanted to say thank you so much for putting this project together.  This is awesome.  I am migrating from MWS to SP-API and I believe I have everything setup correctly according to the Amazon documentation.  I even made the roll modification for “Assume Role” STS“Assume Role” STS that you talked about in your documentation.  When I run it, I'm getting an error that says "Access to requested resource is denied."  I have tried Orders and Product Pricing and they both return the same error.  I stepped through the code and it looks like it's getting an authtoken just fine, but it's getting a 'Forbidden' when executing a rest request to https://sellingpartnerapi-na.amazon.com/products/pricing/v0/competitivePrice.  Do you have a recommendation on how I can can debug this issue on my end?  I'm not even sure where to look to enable these apis.

FikaAmazonAPI.AmazonSpApiSDK.Models.Exceptions.AmazonUnauthorizedException: 'Access to requested resource is denied.'

<img width="374" alt="image" src="https://user-images.githubusercontent.com/32724297/150626510-c3ee2c78-a41c-423e-ac06-fd6570965227.png">

**Comment by abuzuhri** [comment]
Hello @user6116 ,

you welcome ,

Please try to follow all steps here to get the keys needed 
https://developer-docs.amazon.com/sp-api/docs/creating-and-configuring-iam-policies-and-entities

here sample code with pure C# without dependency and simple code to debug 
https://github.com/abuzuhri/Amazon-SP-API-CSharp/tree/main/Others/PureCodeSampleForTest

Check all steps and let me know if you need any help

**Reply by user6116** [reply]
Thanks for that.  I deleted everything I created in AWS to create the user and permission and recreated it to that link again.  Then it worked.  I must have missed a step the first time.  Thanks a lot for the assistance!

**Reply by abuzuhri** [reply]
Hi @user6116 

You are welcome .
